runtime/near-evm-runner/src/builtins.rs

```diff
@@ -0,0 +1,576 @@
+use byteorder::{BigEndian, ByteOrder, LittleEndian, ReadBytesExt};
+use ethereum_types::{Address, H256, U256};
+use num_bigint::BigUint;
+use num_traits::{One, Zero};
+
+use parity_bytes::BytesRef;
+use ripemd160::Digest;
+use std::{
+    cmp::{max, min},
+    io::{self, Cursor, Read},
+    mem::size_of,
+};
+use vm::{MessageCallResult, ReturnData};
+
+pub static COUNT: u64 = 9;
+
+pub fn is_precompile(addr: &Address) -> bool {
+    *addr <= Address::from_low_u64_be(COUNT)
+}
+
+pub fn precompile(id: u64) -> Box<dyn Impl> {
+    match id {
+        1 => Box::new(EcRecover) as Box<dyn Impl>,
+        2 => Box::new(Sha256) as Box<dyn Impl>,
+        3 => Box::new(Ripemd160) as Box<dyn Impl>,
+        4 => Box::new(Identity) as Box<dyn Impl>,
+        5 => Box::new(ModexpImpl) as Box<dyn Impl>,
+        6 => Box::new(Bn128AddImpl) as Box<dyn Impl>,
+        7 => Box::new(Bn128MulImpl) as Box<dyn Impl>,
+        8 => Box::new(Bn128PairingImpl) as Box<dyn Impl>,
+        9 => Box::new(Blake2FImpl) as Box<dyn Impl>,
+        _ => panic!("Invalid builtin ID: {}", id),
+    }
+}
+
+pub fn process_precompile(addr: &Address, input: &[u8]) -> MessageCallResult {
+    let f = precompile(addr.to_low_u64_be());
+    let mut bytes = vec![];
+    let mut output = parity_bytes::BytesRef::Flexible(&mut bytes);
+
+    // mutates bytes
+    f.execute(input, &mut output).expect("No errors in precompiles");
```

Comment:
```suggestion
    f.execute(input, &mut output).expect("Errors in precompiles");
```
because rust would show error message as `Errors in precompiles: {Err from f.execute}`.

Reply:
removed panics